### DISCUSSION TITLE
ExternalRenderTheme + Scoped Storage

### DISCUSSION THREAD

**Discussion by RoProducts** [mapsforge/mapsforge#1474] [Q&A] [answered]
Recent Android Versions are no longer based on java.io.File but the so called Scoped Storage and the use of Uri and DocumentFile. 

In recent Mapsforge version you allow the usage of MapFile using this approach (using a FileInputStream), however ExternalRenderTheme can only be created with a java.io.File asfaik.

Is a solution for Shared Storage planned or already possible with some workaroung ?

Thank you

**Comment by devemux86** [comment, marked as answer]
It is possible opening the themes as zip files with ZipRenderTheme.
See the [MapsforgeMapViewer](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/MapsforgeMapViewer.java) example.
